date.h

```diff
@@ -4140,6 +4162,60 @@ format(const CharT* fmt, const sys_time<Duration>& tp)
                           &abbrev, &offset);
 }
 
+
+// ------------- With ostream as destination -------
+
+template <class CharT, class Traits, class Duration>
+inline
+std::basic_ostream<CharT, Traits>&
+format(std::basic_ostream<CharT, Traits>& os,
+       std::basic_string<CharT, Traits> fmt,
+       const local_time<Duration>& tp)
+{
+    return detail::format(os, std::move(fmt), tp);
+}
+
+
+template <class CharT, class Traits, class Duration>
+inline
+std::basic_ostream<CharT, Traits>&
+format(std::basic_ostream<CharT, Traits>& os,
+       std::basic_string<CharT, Traits> fmt, const sys_time<Duration>& tp)
+{
+    const std::string abbrev("UTC");
+    CONSTDATA std::chrono::seconds offset{0};
+    return detail::format(os, std::move(fmt),
+                          local_time<Duration>{tp.time_since_epoch()}, &abbrev, &offset);
+}
+
+// const CharT* formats
+
+
+template <class CharT, class Traits, class Duration>
+inline
+std::basic_ostream<CharT, Traits>&
+format(std::basic_ostream<CharT, Traits>& os,
+       const CharT* fmt, const local_time<Duration>& tp)
+{
+    return detail::format(os, std::basic_string<CharT>(fmt), tp);
+}
+
+template <class CharT, class Traits, class Duration>
+inline
+std::basic_ostream<CharT, Traits>&
+format(std::basic_ostream<CharT, Traits>& os,
+       const CharT* fmt, const sys_time<Duration>& tp)
+{
+    const std::string abbrev("UTC");
+    CONSTDATA std::chrono::seconds offset{0};
+    return detail::format(os, std::basic_string<CharT>(fmt),
+                          local_time<Duration>{tp.time_since_epoch()},
+                          &abbrev, &offset);
+}
+
+
+// --------- End with ostream as destination -------
```

Comment:
Just made another commit with the MIT copyright inserted - please feel free to use any parts.

In regards with the syntax, my arguments:

1. the way I see, the matter does have something related to the function name and the expectations it creates
2. the `std::time_put` doesn't pretend to offer anything related to manipulators, width and alignment of the entire result - but it's still usable and used. The same goes for the venerable `printf`in which the `fprintf` variant is highly probable used underneath by binding the stream to `stdout`.

Based on the arguments above, can we settle for renaming my proposed `ostream` variant of the `date::format` to something that doesn't create expectations of 'step into the iomanip shoes'? Something that suggest a limited and strictly utilitarian purpose, the way `std::time_put` does - I don't know, something like `date::put` or even 'do_put`?

What I'm after is a function that 'streams' the content of temporal data types straight into an `ostream` at the current position with as little overhead as possible. My current needs require 'datetime', 'date' and `time' data types, which I thought I'd be able to support on top/using `date::` data structs. Even if the library in/by itself is not concerned in anything but marshalling/unmarshalling those types, for the consumers of the library it would be a huge advantage if the structure support calendaring and arithmetic operation - this is why you propose the `date` library yourself, isn't it?

I guess what I'm asking is: please have pity for the gals and guys needing un/marshalling `date` and let them have the streaming straight from (thanks for the imanip) and to streams - no matter the name of the utility function - their needs are legitimate, so perhaps the `date` lib should cater for those needs if it aspires to the status of a standard library.

> You've done a really nice job with this. 
Thanks for the kind words, they mean something for a guy which only a month ago started to look again into C++ after about 7 years of absence - that's me.